packages/core/src/modules/offline/offline-repository.ts

```diff
@@ -0,0 +1,44 @@
+import { type Database } from "@dotkomonline/db"
+import { type Offline, type OfflineId, OfflineSchema } from "@dotkomonline/types"
+import { type Insertable, type Kysely, type Selectable } from "kysely"
+import { type Cursor, orderedQuery } from "../../utils/db-utils"
+
+type OfflineWrite = Insertable<Database["offline"]>
+
+export interface OfflineRepository {
+  getById(id: OfflineId): Promise<Offline | undefined>
+  getAll(take: number, cursor?: Cursor): Promise<Offline[]>
+  create(values: OfflineWrite): Promise<Offline>
+  update(id: OfflineId, data: OfflineWrite): Promise<Offline>
+}
+
+const mapToOffline = (offline: Selectable<Database["offline"]>): Offline => OfflineSchema.parse(offline)
+
+export class OfflineRepositoryImpl implements OfflineRepository {
+  constructor(private readonly db: Kysely<Database>) {}
+
+  private baseOfflineQuery() {
+    return this.db.selectFrom("offline").selectAll("offline")
+  }
```

Comment:
I dont think this shorthand provides much value

Reply:
Agreed, just confusing. ChatGPT ngl